packages/base-map/src/index.js

```diff
@@ -0,0 +1,271 @@
+import React, { Component } from "react";
+import PropTypes from "prop-types";
+import { LayersControl, Map, Popup, TileLayer } from "react-leaflet";
+import { latlngType } from "@opentripplanner/core-utils/lib/types";
+import L from "leaflet";
+
+// eslint-disable-next-line func-names
+L.Evented.addInitHook(function() {
+  if (this) this.singleClickTimeout = null;
+  this.on("click", this.scheduleSingleClick, this);
+  this.on("dblclick dragstart zoomstart", this.cancelSingleClick, this);
+});
+
+L.Evented.include({
+  cancelSingleClick() {
+    // This timeout is key to workaround an issue where double-click events
+    // are fired in this order on some touch browsers: ['click', 'dblclick', 'click']
+    // instead of ['click', 'click', 'dblclick']
+    setTimeout(this.clearSingleClickTimeout.bind(this), 0);
+  },
+
+  scheduleSingleClick(e) {
+    this.clearSingleClickTimeout();
+
+    this.singleClickTimeout = setTimeout(
+      this.fireSingleClick.bind(this, e),
+      this.options.singleClickTimeout || 500
+    );
+  },
+
+  fireSingleClick(e) {
+    // eslint-disable-next-line no-underscore-dangle
+    if (!e.originalEvent._stopped) {
+      this.fire("singleclick", L.Util.extend(e, { type: "singleclick" }));
+    }
+  },
+
+  clearSingleClickTimeout() {
+    if (this.singleClickTimeout !== null) {
+      clearTimeout(this.singleClickTimeout);
+      this.singleClickTimeout = null;
+    }
+  }
+});
+
+/**
+ * The base OpenTripPlanner map on which everything else is rendered.
+ */
+class BaseMap extends Component {
+  componentDidMount() {
+    const lmap = this.refs.map.leafletElement;
+    lmap.options.singleClickTimeout = 250;
+    lmap.on("singleclick", this.onLeftClick);
+  }
+
+  componentDidUpdate() {}
+
+  // remove custom overlays on unmount
+  // TODO: Is this needed? It may have something to do with mobile vs desktop views
+  componentWillUnmount() {
+    const lmap = this.refs.map.leafletElement;
+    lmap.eachLayer(layer => {
+      // Do not inline, there is a 'this' implied.
+      lmap.removeLayer(layer);
+    });
+  }
+
+  onLeftClick = e => {
+    const { onClick } = this.props;
+    if (typeof onClick === "function") onClick(e);
+  };
+
+  render() {
+    const {
+      baseLayers,
+      center,
+      children,
+      maxZoom,
+      popupLocation,
+      popupContent,
+      onPopupClosed,
+      onOverlayAdded,
+      onOverlayRemoved,
+      onViewportChanged,
+      zoom
+    } = this.props;
+
+    // Separate overlay layers into user-controlled (those with a checkbox in
+    // the layer control) and those that are needed by the app (e.g., stop viewer
+    // and itinerary overlay).
+    const userControlledOverlays = [];
+    const fixedOverlays = [];
+    React.Children.toArray(children).forEach(child => {
+      if (child.props.name) userControlledOverlays.push(child);
+      else fixedOverlays.push(child);
+    });
+
+    return (
+      <Map
+        ref="map"
```

Comment:
@binh-dam-ibigroup, were you running into issues with the use of `this.refs.map`? I believe the latest react-leaflet docs say that we should use `this.leafletElement` to access the map, so we can probably remove this `ref`. https://react-leaflet.js.org/docs/en/components